### PR TITLE
feat: introduce `FlowRouteValue`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Introduce `FlowRouteValue` to store flow routes

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Introduced functionality for managing route information for flows, including new structs and a manager for handling flow route operations.

- **Refactor**
  - Removed `StandalonePeerLookupService` and associated code from multiple modules, simplifying the codebase.

### CODE REVIEW

File: src/common/meta/src/ddl/drop_flow.rs [hunk 1]
```diff
@@ -102,18 +104,9 @@ impl DropFlowProcedure {
...
-                .context
-                .peer_lookup_service
-                .flownode(cluster_id, *flownode)
-                .await?
-                .with_context(|| UnexpectedSnafu {
-                    err_msg: "Attempted to drop flow on a node that could not be found. Consider verifying node availability.",
-                })?;
-            let requester = self.context.node_manager.flownode(&peer).await;
+
+        for FlowRouteValue { peer } in &self.data.flow_route_values {
```
Comment (fengjiachun) [inline src/common/meta/src/ddl/drop_flow.rs:108]: Why is the `partition_id` not in `DropRequest`?

File: src/common/meta/src/key/flow/flow_route.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+const FLOW_ROUTE_KEY_PREFIX: &str = "route";
+
+lazy_static! {
+    static ref FLOW_ROUTE_KEY_PATTERN: Regex =
+        Regex::new(&format!("^{FLOW_ROUTE_KEY_PREFIX}/([0-9]+)/([0-9]+)$")).unwrap();
+}
+
+/// The key stores the route info of the flow.
+///
+/// The layout: `__flow/info/{flow_id}/{partition_id}`.
```
Comment (fengjiachun) [inline src/common/meta/src/key/flow/flow_route.rs:42]:
```suggestion
/// The layout: `__flow/flow/{flow_id}/{partition_id}`.
```

File: src/common/meta/src/ddl/drop_flow.rs (hunk 1 above)
Comment (WenyXu) [reply src/common/meta/src/ddl/drop_flow.rs:108]: Good point🥹 I'm going to create an issue to fix this.

Comment (fengjiachun) [reply src/common/meta/src/ddl/drop_flow.rs:108]: OK

**Review by fengjiachun — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**